### ISSUE TITLE
[v2.1.2] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#113962]
This issue is for tracking cherry-picks to the release branch. Following is [release branch](https://github.com/pytorch/pytorch/tree/release/2.1) for the 2.1.2 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 12/1): Cherry-pick post deadline (End of day 5PM PST)
* Phase 2 (after 12/1):  Perform extended integration/stability/performance testing based on Release Candidate builds.

**Only issues that have ‘cherry-picks’ in this tracker will be considered for the release.**

## Cherry-Pick Criteria

**Phase 1 (until 12/1):**

The Releng team relies on the cherry pick process to manage risk to release quality, i.e. by porting a small set of commit from trunk that are "must-have" into the release branch, we limit the change to the minimal to address pressing issues. Thus, not everything a developer land into the trunk will make it into the release. So, please consider the criteria below and follow the cherry picking process. Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent release (e.g. 2.1.0 for 2.1.2 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Low risk critical fixes for: silent correctness, backwards compatibility, crashes, deadlocks, (large) memory leaks
3. Fixes to new features being introduced in 2.1.0 release
4. Documentation improvements
5. Release branch specific changes (e.g. blocking ci fixes, change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @huydhn, @osalpekar, @malfet). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain. 

**Phase 2 (after 12/1):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.1).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.1
    git checkout release/2.1
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/2.1' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.1
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See [HUD 2.1](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.1/1?per_page=50) 

### Versions

2.1.2

**Comment by ringohoffman** [comment]
> Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.0).

The branch linked is `release/2.0`. I think it should be [`release/2.1`](https://github.com/pytorch/pytorch/tree/release/2.1).

**Comment by malfet** [comment]
Link to landed trunk PR :
* https://github.com/pytorch/builder/commit/2a750ebb877695947146a307b22cf27b6c0d0302

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1625

Criteria Category:
* Fixes to new features being introduced in 2.1.0 release (poetry support)

**Comment by malfet** [comment]
Link to landed trunk PR :

* https://github.com/pytorch/pytorch/pull/113763
Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/114416

Criteria Category:

* Regressions from the previous release (2.1.0)

**Comment by malfet** [comment]
Link to landed trunk PR :

* https://github.com/pytorch/pytorch/pull/114418

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/114011

Criteria Category:

* Critical fix to PyTorch behavior as AWS lambda

**Comment by malfet** [comment]
Link to landed trunk PR :

* https://github.com/pytorch/pytorch/pull/113398

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/114419

Criteria Category:

* Fixes crash in Conv backward path on MPS

**Comment by malfet** [comment]
Link to landed trunk PR :

* https://github.com/pytorch/pytorch/pull/113684

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/114420

Criteria Category:

* Critical fix to PyTorch behavior as AWS lambda

**Comment by huydhn** [comment]
Link to landed trunk PR :
* https://github.com/huydhn/builder/commit/c3e0f559d5eeeea3546ec1357112fffa9d5452b2

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1626

Criteria Category:
* Critical CI fix to build PyTorch on Windows for 3.10 and 311

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/114514

Criteria Category:
* Release only, pin tests

**Comment by jcaip** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/105595
* https://github.com/pytorch/pytorch/pull/110420

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/114593

Criteria Category:
* Critical Fixes

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/113055

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/114756

Criteria Category:
* Regression in 2.1.0

**Comment by gdippolito** [comment]
Would it be possible to cherry-pick [this commit](https://github.com/pytorch/pytorch/pull/109363)in 2.1.2? The change is self-contained and would fix a bug that was introduced in 2.1.0. 
I have tested it and the commit can be cherry-picked without conflicts. Thanks 🙏

**Comment by drisspg** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/commit/8556a09d44e3e65ef360ba6a64f724d264ee35cb

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/114837

Criteria Category:
* Fixes a memory regression introduced in 2.1.1 patch fix

**Comment by imzhuhl** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/110150

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/114851

Criteria Category:
* Critical Fixes

**Comment by bhack** [comment]
@atalman  but they are related as the first one is nightly but since july. Are we not in that branch cut since july?

**Comment by huydhn** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/115183

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/115241

Criteria Category:
* Critical fix https://github.com/pytorch/pytorch/issues/113037

**Comment by bhack** [comment]
@atalman I don't have any notification if you post-edit my own messages inline. So please, if you can, reply with a new comment as all other people are doing in almost every Github repo. 

For the others interested, we are discussing details in the mentioned tickets own threads.

**Comment by drisspg** [comment]
> Link to landed trunk PR (if applicable):
> 
> * [Constrain sdpa to fx strides #111721](https://github.com/pytorch/pytorch/pull/111721)
> * [Require less alignment for attn bias #114173](https://github.com/pytorch/pytorch/pull/114173)
> 
> Link to release branch PR:
> 
> * [Require less alignment for attn bias (#114173) #114837](https://github.com/pytorch/pytorch/pull/114837)
> 
> Criteria Category:
> 
> * Fixes a memory regression introduced in 2.1.1 patch fix
> 
> @atalman Looks like cherry pick combines #111721 and #114173 PRs (and also this small tweak #114909)
> 
> @atalman merged
> 
> @atalman: Because of the new issue found [#112577 (comment)](https://github.com/pytorch/pytorch/issues/112577#issuecomment-1852273407) will be reverting this cherry-pick

Follow up here, users in HuggingFace are saying that this inclusion has caused an IMA in one of their models, therefore we are going to revert for the cherry pick but still include in Main while I work to debug this issue.

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/commit/a8e7c98cb95ff97bb30a728c6b2a1ce6bff946eb

Criteria Category:
* Revert of https://github.com/pytorch/pytorch/pull/114837

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/builder/commit/2a4c533360a1aa2dec292baa95e69c7ec6ecd166

Link to release branch PR:
* https://github.com/pytorch/builder/commit/55d40b1b06379fddd0ba14744dc3fdd768530da6

Criteria Category:
* Critical CI fix

---
@atalman merged

**Comment by huydhn** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/test-infra/pull/4803

Link to release branch PR:
* https://github.com/pytorch/test-infra/pull/4804

Criteria Category:
* Critical CI fix to allow building domain RC for both 2.1.2 and 2.2.0 in parallel

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/test-infra/pull/4805

Link to release branch PR:
* https://github.com/pytorch/test-infra/pull/4806

Criteria Category:
* Critical CI fix to allow building domain RC for both 2.1.2 and 2.2.0 in parallel

**Comment by malfet** [comment]
Closing this issue as wheels and conda binaries has been published https://pypi.org/project/torch/2.1.2/ and https://anaconda.org/pytorch/pytorch/files?version=2.1.2

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/115949

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/116063

Criteria Category:
* CI build docker cuda 11.8 for cuda 11.8
